### PR TITLE
Patch save image with tag

### PR DESCRIPTION
**Pull request by draoullig**
Hy,

I found a bug on the command to save images in a tar.
If you push a tag today, it is ignored because the tag is push in queryParam and not in the url (https://docs.docker.com/engine/reference/api/docker_remote_api_v1.19/#get-a-tarball-containing-all-images-in-a-repository).
I made an evolution to patch this problem.

Best regards

Frederic

---

This change is [Reviewable](https://reviewable.io/reviews/docker-java/docker-java/521)

### CODE REVIEW

File: src/main/java/com/github/dockerjava/jaxrs/SaveImageCmdExec.java [hunk 1]
```diff
@@ -22,8 +23,15 @@ public SaveImageCmdExec(WebTarget baseResource, DockerClientConfig dockerClientC
 
     @Override
     protected InputStream execute(SaveImageCmd command) {
-        WebTarget webResource = getBaseResource().path("/images/" + command.getName() + "/get").queryParam("tag",
-                command.getTag());
+    	
+    	// If tag is present, only tar the specific image
```
Comment (KostyaSha) [inline src/main/java/com/github/dockerjava/jaxrs/SaveImageCmdExec.java]: tar?

Comment (KostyaSha) [inline src/main/java/com/github/dockerjava/jaxrs/SaveImageCmdExec.java]: sorry confused with tar, tag :)

**Comment by KostyaSha** [comment]
Seems we would need a test case for it

**Comment by draoullig** [comment]
Hi,

I change my chectstyle environment to be compliant with us.
I add a test integration too.

Best Regards

Frédéric.

**Comment by KostyaSha** [comment]
Missing netty implementation.
Would this change be backward compatible with previous (docker daemon) API versions?

**Comment by KostyaSha** [comment]
As i see `    save      Save an image(s) to a tar archive`
```

Usage:	docker save [OPTIONS] IMAGE [IMAGE...]

Save an image(s) to a tar archive (streamed to STDOUT by default)

  --help             Print usage
  -o, --output       Write to a file, instead of STDOUT
```
Requires only image name. But docker API also allows passing `names` query. So as library we should support all this options.

https://github.com/docker/docker/blob/master/docs/reference/api/docker_remote_api_v1.22.md#get-a-tarball-containing-all-images-in-a-repository contains some https://github.com/docker/docker/blob/master/docs/reference/api/docker_remote_api_v1.22.md#image-tarball-format format. Could you check whether this format is right now and whether it possible implement some integration test (tag something (i.e. busybox) with test specific repository name and verify that returned archive contains tags/images). I think you can add commons-compress into test scope if it missing?

**Comment by marcuslinke** [comment]
@draoullig Ping!

**Comment by draoullig** [comment]
Hi,

I don't have time to contribue at this project. 
I made this patch to solve the bug in a jenkins plugin.
I send you this solution to fix the bug even all the differents uses cases are not covered.

Best regards. 

Frédéric

**Comment by KostyaSha** [comment]
@draoullig thanks, it will take some time to update your changes and we will merge later.

**Comment by KostyaSha** [comment]
retriggering pr build

**Comment by cdancy** [comment]
Any chance we could get this PR massaged and cleaned up and merged?

**Comment by KostyaSha** [comment]
> Any chance we could get this PR massaged and cleaned up and merged?

Too busy on this week, i can press merge and run relese. Could you PR picking this commits?

**Comment by KostyaSha** [comment]
https://github.com/docker-java/docker-java/pull/1155

**Pull request closed without merging**